### PR TITLE
Automatically update push permission state on app resume

### PR DESCRIPTION
**Pull request by evan-masseau**
# Description
Adds automatic tracking of permission state change, by listening for `resume` events from the host app and re-checking permission state and device metadata against the latest push state data.

TODO:
- [x] Add migration notes and readme updates
- [x] Check with test app
- [x] Compile and test edge cases / extended use cases.
- [x] Check on performance implications

# Check List

- [x] Are you changing anything with the public API?
- [x] Are your changes backwards compatible with previous SDK Versions?
- [ ] Have you tested this change on real device?
- [x] Have you added unit test coverage for your changes?
- [x] Have you verified that your changes are compatible with all Android versions the SDK currently supports?

## Changelog / Code Overview

- Attach a state side effect listener to activity lifecycle events
- On resume, if there is a push token in state, refresh overall push state
- Detach listeners on re-initialize to prevent dupes

## Test Plan

- [x] Quick test in an emulator
- [x] Unit tests
- [ ] Extended app testing

## Related Issues/Tickets

#161 - Required for version 3.0.0
CHNL-961

### CODE REVIEW

File: sdk/analytics/src/main/java/com/klaviyo/analytics/Klaviyo.kt [hunk 1]
```diff
@@ -69,6 +69,7 @@ object Klaviyo {
         Registry.get<ApiClient>().startService()
 
         Registry.register<State>(KlaviyoState())
+        Registry.getOrNull<StateSideEffects>()?.detach()
```
Comment (evan-masseau) [inline sdk/analytics/src/main/java/com/klaviyo/analytics/Klaviyo.kt:72]: Because we didn't detach previous listeners, we risked accumulating multiple identical side effect listeners 😬

**Review by jordan-griffin — APPROVED** [review]
